R/hdf5.R

```diff
@@ -1,110 +1,207 @@
-##' @title The `Hdf5MSnExp` Class for MS Data And Meta-Data
-##' @description The `Hdf5MSnExp` class encapsulates data and
-##'     meta-data for mass spectrometry experiments like the `MSnExp`
-##'     and `OnDiskMSnExp` classes. `Hdf5MSnExp` implements, like
-##'     `OnDiskMSnExp`, an *on disk* model, where the raw data (M/Z
-##'     and intensities) are stored on disk (rather than in memory,
-##'     like `MSnExp` data) using, as the name implies, the hdf5 file
-##'     format. See [`MSnExp`] and [`OnDiskMSnExp`] for a more general
-##'     description of the data, slots and operations.
-##'
-##'     Object of the class are currently created with the
-##'     `readHdf5DiskMSData` function. Later, this backend will be
-##'     included in the main `readMSData` function.
-##'
-##' @slot hdf5file `character(1)` containing the hdf5 filename.
-##' @slot hdf5handle A `H5IdComponent` or `NULL`.
-##'
-##' @rdname Hdf5MSnExp-class
-##' @aliases Hdf5MSnExp
-##' @seealso See [`MSnExp`] and [`OnDiskMSnExp`] classes.
-##' @author Laurent Gatto
-##' @md
-##' @examples
-##' f <- msdata::proteomics(pattern = "MS3TMT11", full.names = TRUE)
-##' x <- readHdf5DiskMSData(f)
-##' x
-##'
-##' ## automatically generated hdf5 file
-##' hdf5FileName(x)
-##'
-##' x[[1]]
-##' x[[2]]
-##' x[[10]]
-##' filterMsLevel(x, 3L)[[1]]
-##'
-##' ## clean up session
-##' file.remove(hdf5FileName(x))
-##' validHdf5MSnExp(x) ## not valid anymore
-##' rm(x)
-.Hdf5MSnExp <- setClass("Hdf5MSnExp",
-                        slots = c(hdf5file = "character"),
-                        contains = "OnDiskMSnExp",
-                        prototype = prototype(
-                            new("VersionedBiobase",
-                                versions = c(classVersion("MSnExp"),
-                                             classVersion("OnDiskMSnExp"),
-                                             Hdf5MSnExp = "0.0.1")),
-                            spectraProcessingQueue = list(),
-                            backend = character()))
-
-validHdf5MSnExp <- function(object) {
+validHdf5MSnExp <- function(object, check_md5 = TRUE) {
```

Comment:
validity can be checked with and without checking also whether the **content** of the hdf5 files has changed. The latter is performed by default. `validObject()` will perform a *basic* check with `check_md5 = FALSE`.